src/utils/node_byte_utils.ts

```diff
@@ -0,0 +1,119 @@
+import { BSONError } from '../error';
+import { isAnyArrayBuffer } from '../parser/utils';
+import type { ByteUtils } from './byte_utils';
+
+type NodeJsEncoding = 'base64' | 'hex' | 'utf8' | 'binary';
+type NodeJsBuffer = {
+  alloc: (size: number) => Uint8Array;
+  from(array: number[]): Uint8Array;
+  from(array: Uint8Array): Uint8Array;
+  from(array: ArrayBuffer): Uint8Array;
+  from(array: ArrayBuffer, byteOffset: number, byteLength: number): Uint8Array;
+  from(base64: string, encoding: NodeJsEncoding): Uint8Array;
+  write(string: string, offset: number, length: undefined, encoding: 'utf8'): number;
+  copy(target: Uint8Array, targetStart: number, sourceStart: number, sourceEnd: number): number;
```

Comment:
Not sure how strict you’d like to get with the types here, but these are also prototype methods, not static ones

Reply:
broke up the types to now be a Ctor and an instance, tells more of the truth